calculator/main.py

```diff
@@ -37,14 +62,24 @@ def add_new_operation(calc, operation):
     :param operation: Dict with the single operation to be added.
                       ie: {'add': add_function}
     """
-    pass
+    
+    # Test operation, raise error if invalid
+    if not isinstance(operation, dict):
+        raise InvalidOperation('Given operation is invalid.')
+    if len(operation) != 1:
+        raise InvalidParams('Can only add one operation at a time')
+    
+    # Add operation to calculator
+    operation_name, operation_function = next(iter(operation.items()))
```

Comment:
I really like line 73. 1-liner, and it does not require turning .items() into a list.